src/js/structs/Batch.js

```diff
@@ -55,6 +54,28 @@ class Batch {
    * @param {Object} item
    */
   add(item) {
+    let batchIndex = this.length;
+
+    // As the while is faster then the Array.prototype.forEach and this
+    // function can potentially be called more often we choose while here.
+    while (--batchIndex >= 0) {
+      let {path, value} = this[batchIndex] || {};
+      let hasEqualPaths = item && Array.isArray(path) &&
+        Array.isArray(item.path) && path.join() === item.path.join();
+      if (hasEqualPaths) {
+        // Last entry with same path holds same value,
+        // do not add redundant action
+        if (value === item.value) {
+          return;
+        }
+        // Remove previous if path is the same as current to minimize
+        // number of actions
+        if (batchIndex === this.length - 1) {
+          this.pop();
+        }
+      }
+    }
```

Comment:
The while loop and looking back if the same path had the same value previously is a good idea, but it will not work. Here is one examples: 

```JS
batch.add({value: 'a', path: ['id']});
batch.add({value: 1, path: ['cpu']});
batch.add({value: 'b', path: ['id']});
batch.add({value: 1, path: ['mem']});
batch.add({value: 'a', path: ['id']});
```

I would expect to have 5 events in the batch but I only would have 4 because the last event with the `value: a` and the `path: ['id']` would have been omitted, because it is the same as the first one.

Another example for duplicatable rows:
- value 10, path `['foo', '0', 'bar']`
- add, path `['foo', '1', 'bar']`
- remove, path `['foo', '0', 'bar']`
- value 10, path `['foo', '0', 'bar']`

If we add a `value=10` for the path `['foo', '0', 'bar']` then we add `['foo', 1, 'bar']` then we remove `['foo', '0', 'bar']` and then change the the value to `value=10` for `['foo', '0', 'bar']` which used to be `['foo', '1', 'bar']` this event would be omitted. Which will lead to a false state. 

Also we want to only replace events / omit events if they have a certain type.
example:

- add, path `['foo', '1', 'bar']`
- add, path `['foo', '2', 'bar']`
- add, path `['foo', '3', 'bar']`
- remove, path `['foo', '0', 'bar']`
- remove, path `['foo', '0', 'bar']`
- remove, path `['foo', '0', 'bar']`

In this case we want to have only 1 row left. but by omitting the last 2 rows we are left with 3 rows.

I hope this makes sense in this abstract form.

I would suggest to only do this
```JS
if (item.type === TransactionTypes.SET) {
  let {path, value} = this[this.length - 1] || {};
  let hasEqualPaths = item && Array.isArray(path) &&
    Array.isArray(item.path) && path.join() === item.path.join();
  if(hasEqualPaths && value === item.value) {
    this.pop();
  }
}